src/link/Elf/file.zig

```diff
@@ -240,38 +240,39 @@ pub const File = union(enum) {
         return switch (file) {
             .zig_object => |x| x.updateArSymtab(ar_symtab, elf_file),
             .object => |x| x.updateArSymtab(ar_symtab, elf_file),
-            inline else => unreachable,
+            else => unreachable,
         };
     }
 
     pub fn updateArStrtab(file: File, allocator: Allocator, ar_strtab: *Archive.ArStrtab) !void {
-        const path = switch (file) {
-            .zig_object => |x| x.path,
-            .object => |x| x.path,
-            inline else => unreachable,
-        };
-        const state = switch (file) {
-            .zig_object => |x| &x.output_ar_state,
-            .object => |x| &x.output_ar_state,
-            inline else => unreachable,
-        };
-        if (path.len <= Archive.max_member_name_len) return;
-        state.name_off = try ar_strtab.insert(allocator, path);
+        switch (file) {
+            .zig_object => |zo| {
+                const basename = zo.basename;
+                if (basename.len <= Archive.max_member_name_len) return;
+                zo.output_ar_state.name_off = try ar_strtab.insert(allocator, basename);
+            },
+            .object => |o| {
+                const basename = std.fs.path.basename(o.path.sub_path);
+                if (basename.len <= Archive.max_member_name_len) return;
+                o.output_ar_state.name_off = try ar_strtab.insert(allocator, basename);
+            },
```

Comment:
@kubkon archive files cannot have path components in them, so I do basename here. I believe that is causing the test to fail.

Reply:
I think you are absolutely right.